### DISCUSSION TITLE
Is it possible to combine to have consent and preferences in a single modal?

### DISCUSSION THREAD

**Discussion by alexanderadam** [orestbida/cookieconsent#680] [Q&A]
So instead of having a button that opens the settings modal just a single modal that's like

```
headline

description description description description description description description description 

necessary cookies [switch]
functionality cookies [switch]
analytics cookies [switch]

[save selection button] [accept all]
```

**Comment by orestbida** [comment]
@alexanderada, you could technically configure the plugin to directly show the preferences modal, instead of the small popup.

Alternatively, you could design your own consent modal ([demo](https://stackblitz.com/edit/cookieconsent-v3-core?file=index.js)).
